### ISSUE TITLE
package-lock modified but not committed to git repo during "npm version [major|minor|patch]

### ISSUE THREAD

**Issue by RoboPhred** [npm/npm#17707]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [x] Other (_see below for feature requests_):

#### What's going wrong?

When using "npm version [major | minor | patch]" on a project in a git repo, npm will bump the version then commit the package.json changes to git.  However, package-lock.json also gets updated with the bumped version but is not included in the commit.  This leaves the git repo with unstaged changes after a version bump, requiring a separate commit or amend.

#### How can the CLI team reproduce the problem?

Create a node project inside a git repo.  Install a dependency to cause the creation of a package-lock.json.  Run "npm version patch", and observe the creation of a new git commit, and the presence of package-lock as an unstaged change under "git status".

### supporting information:

 - `npm -v` prints: 5.1.0
 - `node -v` prints: 6.9.1
 - `npm config get registry` prints: (internal corp proxy registry)
 - Windows, OS X/macOS, or Linux?: Windows 10

**Comment by stuplum** [comment]
Additionally, if you were to just amend the package-lock.json file to the previous commit, the tag which is created is pointing to the wrong hash.

**Comment by TehShrike** [comment]
I've observed this issue on macOS.

**Comment by stuplum** [comment]
Sorry I should have maybe added:

**supporting information:**

- `npm -v` prints: 5.1.0
- `node -v` prints: 8.1.3
- `npm config get registry` prints: undefined
- Windows, OS X/macOS, or Linux?: macOS 10.12.5

**Comment by markpeterfejes** [comment]
The problem originates from the git index.lock. The npm command wants to add both the package-lock(or the shrinkwrap) and the package.json concurrently, which causes an error on the git side. The error is silent because in the version.js (lib) the addLocalFile function has the parameter true for ignoreFailure.

Relevant code from `/lib/version.js`: 

```javascript
function _commit (version, localData, cb) {
  const options = { env: process.env }
  const message = npm.config.get('message').replace(/%s/g, version)
  const sign = npm.config.get('sign-git-tag')
  const flag = sign ? '-sm' : '-am'
  BB.join(
    addLocalFile('package.json', options, false),
    localData.hasShrinkwrap &&
    addLocalFile('npm-shrinkwrap.json', options, true),
    localData.hasPackageLock &&
    addLocalFile('package-lock.json', options, true)
  ).then(() => {
    return git.exec([ 'commit', '-m', message ], options)
  }).then(() => {
    if (!localData.existingTag) {
      return git.exec([
        'tag', npm.config.get('tag-version-prefix') + version,
        flag, message
      ], options)
    }
  }).nodeify(cb)
}
```
In my opinion and easy solution would be either commiting with the -a flag set, or running the functions in the BB.join sequentially. I can create a pull request if it would help.

This is the error it gives if we set the ignoreFailure false:

```
npm version minor -m "test commit"
v1.10.0
npm ERR! code 128
npm ERR! Command failed: git -c core.longpaths=true add ......\package-lock.json
npm ERR! fatal: Unable to create '......./.git/index.lock': File exists.
npm ERR!
npm ERR! Another git process seems to be running in this repository, e.g.
npm ERR! an editor opened by 'git commit'. Please make sure all processes
npm ERR! are terminated then try again. If it still fails, a git process
npm ERR! may have crashed in this repository earlier:
npm ERR! remove the file manually to continue.
npm ERR!

npm ERR! A complete log of this run can be found in:
npm ERR!     ...
```

**Comment by RoboPhred** [comment]
This seems to be fixed by #17742.  I have not seen this issue for some time over a few versions, and not at my current version of 5.5.1.

Over and out.
